### DISCUSSION TITLE
Adding multiple Beatmaps to collections - more ctrl + click / quality of life improvements

### DISCUSSION THREAD

**Discussion by Zelyson** [ppy/osu#31049] [Ideas]
So I've seen some similar discussions and just wanted to combine them here and give some general input for how this could be done. The first one is also a general quality of life improvement for managing songs.

### Adding Multiple Beatmaps to collections
Being able to add multiple beatmaps to a collection would be amazing as currently you can only add one song at a time. 
I have two ideas for this:

1.  Holding ctrl while clicking on a beatmap allows for selecting multiple maps. With this you could search for and select multiple beatmaps and manage them via a right click (delete, hide, add to collection, etc...). Just clicking without holding ctrl resets the selection while searching / changing the sorting order would keep songs selected. Selected beatmaps could be highlighted the same way a single map is when you click on it. 
This would make it much easier to manage a large number of beatmaps. 
2. Adding more options in the 'manage collections' section (right click on a map, collections > manage...). This is the far more work intensive option but it could open up interesting ways to manage and edit collections. Right clicking on a collection could create a popup/dropdown with options like: 
    - Manage songs - open a searchable list do add songs in one click (left click for add, right click to remove), or have buttons to add/remove songs
    - merge - merge multiple collections into one, maybe by first selecting them with ctrl+click
    - open - to directly open the collection to play songs

Would love some input on this because especially the 'ctrl+click to select multiple songs' seems like a relatively simple feature with big impact.

**Comment by WitherFlower** [comment]
Your idea makes me think of a menu that would look like this. Think of it as a compact version of song select to allow showing more maps at once, but with the same capabilities.

![image](https://github.com/user-attachments/assets/77908ccc-c969-4842-8ac6-98b1b90cfb51)

**Comment by Zelyson** [comment]
Yes, exactly what I imagined. This could open from a "drop down > edit" appearing after right clicking (ignore other options :) ) or an extra button on the side:

![manage - edit2](https://github.com/user-attachments/assets/98fbb77a-a006-4abf-bf69-3cb92bc3da63)
Attribution for the pencil icon: <a href="https://www.flaticon.com/free-icons/edit" title="edit icons">Edit icons created by Pixel perfect - Flaticon</a>

Creating a new collection could also open up a menu like this.
